### DISCUSSION TITLE
windows to linux

### DISCUSSION THREAD

**Discussion by prateek98765** [strongswan/strongswan#952] [Q&A]
Hi, I want to create, host to host connection between windows server and Linux server using ikev2,

i am getting the following error

13[IKE] peer requested virtual IP %any
13[IKE] no virtual IP found for %any requested by '10.10.10.2'
13[IKE] peer requested virtual IP %any6
13[IKE] no virtual IP found for %any6 requested by '10.10.10.2'
13[IKE] no virtual IP found, sending INTERNAL_ADDRESS_FAILURE
13[IKE] configuration payload negotiation failed, no CHILD_SA built
13[IKE] failed to establish CHILD_SA, keeping IKE_SA
13[ENC] generating IKE_AUTH response 1 [ IDr AUTH N(INT_ADDR_FAIL) ]
13[NET] sending packet: from 10.10.10.4[500] to 10.10.10.2[500] (128 bytes)

this error on windows :
---------------------------
Routing and Remote Access
---------------------------
An error occurred during connection of the interface.
Error in assigning inner IP address to initiator in tunnel mode.

---------------------------
OK   
---------------------------

swanctl conf file

 connections {
        specv-host {
            local_addrs = %any
            remote_addrs =%any
            proposals = aes256-sha256-prfsha256-modp1024, aes128-sha256-prfsha256-modp1024
            rekey_time = 0
            reauth_time = 0
            local {
                auth = psk
            }
            remote {
                auth = psk
            }
            children {
                specv-net {
                    rekey_time = 0
                    rand_time = 0
                    esp_proposals = aes128-sha1-ecp256,aes128gcm16-ecp256
                    start_action = trap
                }
            }
        }
    }

secrets {
        ike-psk {
            secret = "12345"
        }
    }

**Comment by strongX509** [comment]
You have to define an IPv4 pool and [optionally] an IPv6 pool in in a special **pools** section in **swanctl.conf**. Then reference the pools via the **pools** attribute in the connection definition. For details see

https://docs.strongswan.org/strongswan-docs/5.9/interop/windowsMachineServerConf.html
